app/javascript/components/layout/ActionTable.js

```diff
@@ -0,0 +1,276 @@
+   */
+  handleCheckboxChange = (e, row) => {
+    const checked = e.target.checked;
+
+    // If row is selected and wasn't previously, add it to the selected rows.
+    if (checked && !this.props.selectedRows.includes(row)) {
+      const selectedRows = [...this.props.selectedRows, row];
+
+      // Update select all if all displayed rows are checked.
+      if (selectedRows >= this.props.entries) {
+        this.setState({ selectAll: true });
+      }
+
+      // Call parent handler
+      this.props.handleSelect(selectedRows);
+    } else {
+      // Otherwise if it was unchecked, remove it from the selected rows and toggle off select all if applicable.
+      const selectedRows = [...this.props.selectedRows];
+      const index = selectedRows.indexOf(row);
+      selectedRows.splice(index, 1);
+
+      // Update select all if not all displayed rows are checked.
+      this.setState({ selectAll: false });
+
+      // Call parent handler
+      this.props.handleSelect(selectedRows);
+    }
+  };
+
+  /**
+   * Called when the checkbox in the table header is clicked and either selects all rows
+   * or deselects all rows based on the current state.
+   */
+  toggleSelectAll = () => {
+    this.setState(
+      state => {
+        return {
+          selectAll: !state.selectAll,
+        };
+      },
+      () => {
+        const selectedRows = this.state.selectAll ? [...Array(this.props.rowData.length).keys()] : [];
+
+        // Call parent handler
+        this.props.handleSelect(selectedRows);
+      }
+    );
+  };
+
+  /**
+   * Called when the edit button a row is clicked. Calls passed in handler (if any).
+   * @param {Number} row - Row that the edit button was clicked on.
+   */
+  handleEditClick = row => {
+    if (this.props.isEditable && this.props.handleEdit) {
+      this.props.handleEdit(row);
+    } else if (this.props.isEditable) {
+      console.log('Please provide a handler function in component props for editing.');
+    }
+  };
+
+  /**
+   * Renders the header element of the table for a given field with
+   * optional sorting functionality and a toolip.
+   *
+   * @param {String} field - Field in the data this column corresponds to.
+   * @param {String} label - Display label for the column.
+   * @param {Boolean} sortable - True if this column should be sortable and false otherwise.
+   * @param {String} tooltip - Text for the tooltip (if any).
+   */
+  renderTableHeader = (field, label, sortable, tooltip) => {
+    return (
+      <th
+        key={field}
+        onClick={() => {
+          if (sortable) {
+            this.handleSortClick(field);
+          }
+        }}
+        className={sortable ? 'pr-3' : ''}
+        style={{ cursor: sortable ? 'pointer' : 'default' }}>
+        {sortable && (
+          <React.Fragment>
+            <div style={{ position: 'relative' }}>
+              <i className="fas fa-sort float-right my-1" style={{ color: '#b8b8b8', position: 'absolute', right: '-12px' }}></i>
```

Comment:
Possible to move these styles to a dedicated admin.scss stylesheet?

Reply:
Yep, that was on my TODO.

Reply:
Done.